diff -Nur old/maildir/configure.in new/maildir/configure.in
--- old/maildir/configure.in	2003-01-30 06:33:31.000000000 +0000
+++ new/maildir/configure.in	2003-12-15 11:15:36.000000000 +0000
@@ -59,20 +59,7 @@
 dnl Checks for library functions.
 AC_CHECK_HEADER(fam.h, :, :)
 AC_CHECK_FUNCS(symlink readlink strcasecmp)
-AC_CHECK_LIB(fam, FAMOpen,  [
-			LIBFAM=-lfam
-			AC_DEFINE_UNQUOTED(HAVE_FAM,1,
-			[ Whether libfam.a is available ])
-
-			AC_CHECK_HEADER(fam.h, : , [
-AC_MSG_WARN([[The development header files and libraries for fam,]])
-AC_MSG_WARN([[the File Alteration Monitor, are not installed.]])
-AC_MSG_WARN([[You appear to have the FAM runtime libraries installed,]])
-AC_MSG_WARN([[so you need to simply install the additional development]])
-AC_MSG_WARN([[package for your operating system.]])
-AC_MSG_ERROR([[FAM development libraries not found.]]) ]
-					)
-		  ])
+AC_CHECK_LIB(fam, FAMOpen, ,)
 
 AC_SUBST(LIBFAM)
 
