diff -Nur old/courier/imapd.rc.in new/courier/imapd.rc.in
--- old/courier/imapd.rc.in	2005-07-02 02:11:16.000000000 +0000
+++ new/courier/imapd.rc.in	2005-08-16 20:48:40.869430152 +0000
@@ -48,6 +48,7 @@
 	;;
 stop)
 	@courierlogger@ -pid=$PIDFILE -stop
+	rm -f $PIDFILE
 	;;
 restart)
 	@courierlogger@ -pid=$PIDFILE -restart
diff -Nur old/courier/imapd-ssl.rc.in new/courier/imapd-ssl.rc.in
--- old/courier/imapd-ssl.rc.in	2005-07-02 23:51:38.000000000 +0000
+++ new/courier/imapd-ssl.rc.in	2005-08-16 20:49:03.195036144 +0000
@@ -49,6 +49,7 @@
 	;;
 stop)
 	@courierlogger@ -pid=$SSLPIDFILE -stop
+	rm -f $SSLPIDFILE
 	;;
 esac
 exit 0
diff -Nur old/courier/module.esmtp/esmtpd.in new/courier/module.esmtp/esmtpd.in
--- old/courier/module.esmtp/esmtpd.in	2005-02-16 23:42:13.000000000 +0000
+++ new/courier/module.esmtp/esmtpd.in	2005-07-03 22:13:30.440224352 +0000
@@ -38,6 +38,7 @@
 	;;
 stop)
 	${sbindir}/couriertcpd -pid=$PIDFILE -stop
+	rm -f $PIDFILE
 	exit 0
 	;;
 restart)
diff -Nur old/courier/module.esmtp/esmtpd-ssl.in new/courier/module.esmtp/esmtpd-ssl.in
--- old/courier/module.esmtp/esmtpd-ssl.in	2005-02-16 23:42:13.000000000 +0000
+++ new/courier/module.esmtp/esmtpd-ssl.in	2005-07-03 22:13:01.767583256 +0000
@@ -28,6 +28,7 @@
 	;;
 stop)
 	${sbindir}/couriertcpd -pid=$SSLPIDFILE -stop
+	rm -f $SSLPIDFILE
 	exit 0
 	;;
 restart)
diff -Nur old/courier/pop3d.in new/courier/pop3d.in
--- old/courier/pop3d.in	2005-07-02 02:11:16.000000000 +0000
+++ new/courier/pop3d.in	2005-08-16 20:49:27.358362760 +0000
@@ -42,6 +42,7 @@
 	;;
 stop)
 	@courierlogger@ -pid=$PIDFILE -stop
+	rm -f $PIDFILE
 	exit 0
 	;;
 restart)
diff -Nur old/courier/pop3d-ssl.in new/courier/pop3d-ssl.in
--- old/courier/pop3d-ssl.in	2005-07-02 23:51:38.000000000 +0000
+++ new/courier/pop3d-ssl.in	2005-08-16 20:49:16.509012112 +0000
@@ -52,6 +52,7 @@
 	;;
 stop)
 	@courierlogger@ -pid=$SSLPIDFILE -stop
+	rm -f $SSLPIDFILE
 	;;
 restart)
 	@courierlogger@ -pid=$SSLPIDFILE -restart
diff -Nur old/courier/webmaild.in new/courier/webmaild.in
--- old/courier/webmaild.in	2005-02-16 23:42:13.000000000 +0000
+++ new/courier/webmaild.in	2005-07-03 22:15:29.401139544 +0000
@@ -42,6 +42,8 @@
 stop)
 	@courierlogger@ -pid=${PIDFILE} -stop
 	@courierlogger@ -pid=${PIDFILE}.pcp -stop
+	rm -f ${PIDFILE}
+	rm -f ${PIDFILE}.pcp
 	;;
 reload)
 	@courierlogger@ -pid=${PIDFILE} -restart
diff -Nur old/courier.sysvinit.in new/courier.sysvinit.in
--- old/courier.sysvinit.in	2005-05-12 15:21:52.000000000 +0000
+++ new/courier.sysvinit.in	2005-07-03 22:12:14.255806144 +0000
@@ -8,6 +8,9 @@
 #
 #
 
+# Source function library
+. /etc/rc.d/init.d/functions
+
 prefix="@prefix@"
 exec_prefix="@exec_prefix@"
 sysconfdir="@sysconfdir@"
@@ -24,6 +27,11 @@
 
 case "$1" in
 start)
+	if [ -e /var/lock/subsys/courier ]; then
+	    msg_already_running Courier
+	    exit 0
+	fi
+
         cd /
         # Start daemons.
         touch /var/lock/subsys/courier
@@ -190,6 +198,8 @@
 	echo ""
         ;;
 stop)
+    if [ -e /var/lock/subsys/courier ]; then
+
         echo -n "Stopping Courier mail server:"
 
 	if test -x ${sbindir}/imapd
@@ -244,6 +254,11 @@
 	${sbindir}/courierfilter stop
 	echo " courierfilter"
         rm -f /var/lock/subsys/courier
+
+    else
+	msg_not_running Courier
+    fi
+
 	;;
 restart)
 	$0 stop
