diff -ur courier-0.64.2.orig//courier/cmlm.h courier-0.64.2/courier/cmlm.h
--- courier-0.64.2.orig//courier/cmlm.h	2007-03-11 05:44:43.000000000 +0100
+++ courier-0.64.2/courier/cmlm.h	2010-05-08 23:28:42.329517182 +0200
@@ -22,6 +22,7 @@
 #include	<functional>
 #include	<cctype>
 
+#define SBINDIR "/usr/sbin"
 #define	TMP	"tmp"
 #define	TMPLOCK	"tmp.lock"
 
@@ -56,7 +57,7 @@
 
 #include	"bindir.h"
 
-#define	SENDMAIL	BINDIR "/sendmail"
+#define	SENDMAIL	SBINDIR "/sendmail"
 #define	REFORMIME	BINDIR "/reformime"
 #define	MAXRCPTS	20
 
diff -ur courier-0.64.2.orig//courier/Makefile.am courier-0.64.2/courier/Makefile.am
--- courier-0.64.2.orig//courier/Makefile.am	2009-12-19 01:59:59.000000000 +0100
+++ courier-0.64.2/courier/Makefile.am	2010-05-08 23:28:42.329517182 +0200
@@ -49,7 +49,7 @@
 extrabin=maildrop lockmail reformail reformime mailbot mimegpg \
 	imapd makemime deliverquota maildirmake maildiracl maildirkw
 
-bin_PROGRAMS=cancelmsg mailq sendmail \
+bin_PROGRAMS=cancelmsg mailq \
 	testmxlookup dotforward \
 	couriermlm webmlmd \
 	courier-config @couriertls@ @EXTRABIN@
@@ -63,7 +63,7 @@
 sbin_SCRIPTS=@SBINSCRIPTS@
 EXTRA_SCRIPTS=
 
-sbin_PROGRAMS=courier showconfig showmodules aliaslookup \
+sbin_PROGRAMS=courier showconfig showmodules aliaslookup sendmail \
 	@COURIERLDAPALIASD@ @EXTRASBIN@
 
 sysconf_DATA=pop3d.authpam imapd.authpam webmail.authpam imapd.cnf pop3d.cnf \
