diff -Nur old/courier/courierd.dist.in new/courier/courierd.dist.in
--- old/courier/courierd.dist.in	2002-03-15 19:17:18.000000000 +0000
+++ new/courier/courierd.dist.in	2004-02-20 12:25:45.000000000 +0000
@@ -76,7 +76,7 @@
 #  Yes, it's two lines long, with an embedded newline.  Of course, you can use
 #  any default local mail delivery instruction in place of ./Maildir.
 
-DEFAULTDELIVERY=./Maildir
+DEFAULTDELIVERY=./Mail/Maildir
 
 ##NAME: MAILDROPDEFAULT:0
 #
@@ -84,7 +84,7 @@
 #  the location of the default mailbox.  You should not change this setting
 #  unless you REALLY know what you're doing.
 
-MAILDROPDEFAULT=./Maildir
+MAILDROPDEFAULT=./Mail/Maildir
 
 ##NAME: ESMTP_CORK:0
 #
diff -Nur old/courier/imapd.rc.in new/courier/imapd.rc.in
--- old/courier/imapd.rc.in	2003-11-29 02:16:15.000000000 +0000
+++ new/courier/imapd.rc.in	2004-02-20 12:25:45.000000000 +0000
@@ -49,7 +49,7 @@
 			-maxprocs=$MAXDAEMONS -maxperip=$MAXPERIP \
 			-pid=$PIDFILE $TCPDOPTS \
 			$PORT @libexecdir@/courier/imaplogin $LIBAUTHMODULES \
-				@bindir@/imapd Maildir"
+				@bindir@/imapd Mail/Maildir"
 	;;
 stop)
 	@sbindir@/couriertcpd -pid=$PIDFILE -stop
diff -Nur old/courier/imapd-ssl.rc.in new/courier/imapd-ssl.rc.in
--- old/courier/imapd-ssl.rc.in	2003-11-29 02:16:15.000000000 +0000
+++ new/courier/imapd-ssl.rc.in	2004-02-20 12:25:45.000000000 +0000
@@ -50,7 +50,7 @@
 			-pid=$SSLPIDFILE $TCPDOPTS \
 			$SSLPORT $COURIERTLS -server -tcpd \
 				@libexecdir@/courier/imaplogin $LIBAUTHMODULES \
-				@bindir@/imapd Maildir"
+				@bindir@/imapd Mail/Maildir"
 	;;
 stop)
 	@sbindir@/couriertcpd -pid=$SSLPIDFILE -stop
diff -Nur old/courier/pop3d.in new/courier/pop3d.in
--- old/courier/pop3d.in	2003-11-28 01:04:59.000000000 +0000
+++ new/courier/pop3d.in	2004-02-20 12:25:45.000000000 +0000
@@ -77,4 +77,4 @@
 		-maxprocs=$MAXDAEMONS -maxperip=$MAXPERIP \
 		-pid=$PIDFILE $TCPDOPTS -address=$ADDRESS $PORT \
 		@libexecdir@/courier/courierpop3login $AUTHMODULELIST \
-			@libexecdir@/courier/courierpop3d Maildir"
+			@libexecdir@/courier/courierpop3d Mail/Maildir"
diff -Nur old/courier/pop3d-ssl.in new/courier/pop3d-ssl.in
--- old/courier/pop3d-ssl.in	2003-11-29 01:37:34.000000000 +0000
+++ new/courier/pop3d-ssl.in	2004-02-20 12:25:45.000000000 +0000
@@ -54,7 +54,7 @@
 			-pid=$SSLPIDFILE $TCPDOPTS \
 			$SSLPORT $COURIERTLS -server -tcpd \
 				$libexecdir/courier/courierpop3login $LIBAUTHMODULES \
-				$libexecdir/courier/courierpop3d Maildir"
+				$libexecdir/courier/courierpop3d Mail/Maildir"
 	;;
 stop)
 	@sbindir@/couriertcpd -pid=$SSLPIDFILE -stop
diff -Nur old/imap/imapd.dist.in new/imap/imapd.dist.in
--- old/imap/imapd.dist.in	2004-01-25 05:40:04.000000000 +0000
+++ new/imap/imapd.dist.in	2004-03-16 17:45:49.000000000 +0000
@@ -371,4 +371,4 @@
 #
 # MAILDIRPATH - directory name of the maildir directory.
 #
-MAILDIRPATH=Maildir
+MAILDIRPATH=Mail/Maildir
diff -Nur old/imap/pop3d.dist.in new/imap/pop3d.dist.in
--- old/imap/pop3d.dist.in	2004-01-25 05:40:04.000000000 +0000
+++ new/imap/pop3d.dist.in	2004-03-16 17:45:56.000000000 +0000
@@ -145,4 +145,4 @@
 #
 # MAILDIRPATH - directory name of the maildir directory.
 #
-MAILDIRPATH=Maildir
+MAILDIRPATH=Mail/Maildir
diff -Nur old/pcp/pcpd.c new/pcp/pcpd.c
--- old/pcp/pcpd.c	2004-05-02 14:59:42.000000000 +0000
+++ new/pcp/pcpd.c	2004-05-20 15:55:56.000000000 +0000
@@ -2187,10 +2187,10 @@
 		exit(1);
 	}
 
-	if (chdir(a->maildir && *a->maildir ? a->maildir:"Maildir") < 0)
+	if (chdir(a->maildir && *a->maildir ? a->maildir:"Mail/Maildir") < 0)
 	{
 		free(token);
-		fprintf(stderr, "NOTICE: chdir(Maildir) failed: %s\n",
+		fprintf(stderr, "NOTICE: chdir(Mail/Maildir) failed: %s\n",
 			strerror(errno));
 		maildir_cache_cancel();
 		exit(1);
diff -Nur old/webmail/auth.c new/webmail/auth.c
--- old/webmail/auth.c	2004-05-09 03:33:22.000000000 +0000
+++ new/webmail/auth.c	2004-05-20 15:54:25.000000000 +0000
@@ -82,7 +82,7 @@
 	if (!maildir || !*maildir)
 		maildir=getenv("MAILDIRPATH");
 	if (!maildir || !*maildir)
-		maildir="Maildir";
+		maildir="Mail/Maildir";
 	if (chdir(maildir))	return (-1);
 	maildir_loginexec();
 	return (0);
@@ -111,7 +111,7 @@
 struct stat stat_buf;
 
 	if (!p || !*p)	p=getenv("MAILDIRPATH");
-	if (!p || !*p)	p="Maildir";
+	if (!p || !*p)	p="Mail/Maildir";
 
 	buf=malloc(strlen(h)+strlen(p)+sizeof(PASSFILE)+100);
 
diff -Nur old/webmail/sqwebmaild.dist.in new/webmail/sqwebmaild.dist.in
--- old/webmail/sqwebmaild.dist.in	2004-05-09 03:33:22.000000000 +0000
+++ new/webmail/sqwebmaild.dist.in	2004-05-28 10:50:17.000000000 +0000
@@ -32,7 +32,7 @@
 #
 # MAILDIRPATH - directory name of the maildir directory.
 
-MAILDIRPATH=Maildir
+MAILDIRPATH=Mail/Maildir
 
 ##NAME: AUTHMODULES:0
 #
diff -Nur old/webmail/sqwebmail.h new/webmail/sqwebmail.h
--- old/webmail/sqwebmail.h	2004-02-29 20:50:13.000000000 +0000
+++ new/webmail/sqwebmail.h	2004-04-15 10:58:47.000000000 +0000
@@ -25,7 +25,7 @@
 
 /* Location of the user's Maildir */
 
-#define USER_DIR	"Maildir"
+#define USER_DIR	"Mail/Maildir"
 
 /* For PAM-based authentication */
 
