diff -urN courier-0.64.2.orig//courier/imapd.rc.in courier-0.64.2/courier/imapd.rc.in
--- courier-0.64.2.orig//courier/imapd.rc.in	2005-07-02 04:11:16.000000000 +0200
+++ courier-0.64.2/courier/imapd.rc.in	2010-05-08 23:30:04.683888399 +0200
@@ -48,6 +48,7 @@
 	;;
 stop)
 	@courierlogger@ -pid=$PIDFILE -stop
+	rm -f $PIDFILE
 	;;
 restart)
 	@courierlogger@ -pid=$PIDFILE -restart
diff -urN courier-0.64.2.orig//courier/imapd-ssl.rc.in courier-0.64.2/courier/imapd-ssl.rc.in
--- courier-0.64.2.orig//courier/imapd-ssl.rc.in	2005-07-03 00:53:19.000000000 +0200
+++ courier-0.64.2/courier/imapd-ssl.rc.in	2010-05-08 23:30:04.685962634 +0200
@@ -49,6 +49,7 @@
 	;;
 stop)
 	@courierlogger@ -pid=$SSLPIDFILE -stop
+	rm -f $SSLPIDFILE
 	;;
 esac
 exit 0
diff -urN courier-0.64.2.orig//courier/module.esmtp/esmtpd.in courier-0.64.2/courier/module.esmtp/esmtpd.in
--- courier-0.64.2.orig//courier/module.esmtp/esmtpd.in	2009-07-30 04:29:54.000000000 +0200
+++ courier-0.64.2/courier/module.esmtp/esmtpd.in	2010-05-08 23:30:04.695962596 +0200
@@ -38,6 +38,7 @@
 	;;
 stop)
 	${sbindir}/couriertcpd -pid=$PIDFILE -stop
+	rm -f $PIDFILE
 	exit 0
 	;;
 restart)
diff -urN courier-0.64.2.orig//courier/module.esmtp/esmtpd-ssl.in courier-0.64.2/courier/module.esmtp/esmtpd-ssl.in
--- courier-0.64.2.orig//courier/module.esmtp/esmtpd-ssl.in	2009-07-30 04:29:54.000000000 +0200
+++ courier-0.64.2/courier/module.esmtp/esmtpd-ssl.in	2010-05-08 23:30:04.695962596 +0200
@@ -28,6 +28,7 @@
 	;;
 stop)
 	${sbindir}/couriertcpd -pid=$SSLPIDFILE -stop
+	rm -f $SSLPIDFILE
 	exit 0
 	;;
 restart)
diff -urN courier-0.64.2.orig//courier/pop3d.in courier-0.64.2/courier/pop3d.in
--- courier-0.64.2.orig//courier/pop3d.in	2005-07-02 04:11:16.000000000 +0200
+++ courier-0.64.2/courier/pop3d.in	2010-05-08 23:30:04.695962596 +0200
@@ -42,6 +42,7 @@
 	;;
 stop)
 	@courierlogger@ -pid=$PIDFILE -stop
+	rm -f $PIDFILE
 	exit 0
 	;;
 restart)
diff -urN courier-0.64.2.orig//courier/pop3d-ssl.in courier-0.64.2/courier/pop3d-ssl.in
--- courier-0.64.2.orig//courier/pop3d-ssl.in	2005-07-03 01:24:34.000000000 +0200
+++ courier-0.64.2/courier/pop3d-ssl.in	2010-05-08 23:30:04.695962596 +0200
@@ -52,6 +52,7 @@
 	;;
 stop)
 	@courierlogger@ -pid=$SSLPIDFILE -stop
+	rm -f $SSLPIDFILE
 	;;
 restart)
 	@courierlogger@ -pid=$SSLPIDFILE -restart
diff -urN courier-0.64.2.orig//courier/webmaild.in courier-0.64.2/courier/webmaild.in
--- courier-0.64.2.orig//courier/webmaild.in	2007-07-01 00:07:21.000000000 +0200
+++ courier-0.64.2/courier/webmaild.in	2010-05-08 23:30:04.695962596 +0200
@@ -43,6 +43,8 @@
 stop)
 	@courierlogger@ -pid=${PIDFILE} -stop
 	@courierlogger@ -pid=${PIDFILE}.pcp -stop
+	rm -f ${PIDFILE}
+	rm -f ${PIDFILE}.pcp
 	;;
 reload)
 	@courierlogger@ -pid=${PIDFILE} -restart
diff -urN courier-0.64.2.orig//courier.sysvinit.in courier-0.64.2/courier.sysvinit.in
--- courier-0.64.2.orig//courier.sysvinit.in	2009-08-29 04:51:52.000000000 +0200
+++ courier-0.64.2/courier.sysvinit.in	2010-05-08 23:30:04.695962596 +0200
@@ -8,6 +8,9 @@
 #
 #
 
+# Source function library
+. /etc/rc.d/init.d/functions
+
 prefix="@prefix@"
 exec_prefix="@exec_prefix@"
 datarootdir="@datarootdir@"
@@ -25,6 +28,11 @@
 
 case "$1" in
 start)
+	if [ -e /var/lock/subsys/courier ]; then
+	    msg_already_running Courier
+	    exit 0
+	fi
+
         cd /
         # Start daemons.
         touch /var/lock/subsys/courier
@@ -200,6 +208,8 @@
 	echo ""
         ;;
 stop)
+    if [ -e /var/lock/subsys/courier ]; then
+
         echo -n "Stopping Courier mail server:"
 
 	if test -x ${bindir}/webmlmd
@@ -260,6 +270,11 @@
 	${sbindir}/courierfilter stop
 	echo " courierfilter"
         rm -f /var/lock/subsys/courier
+
+    else
+	msg_not_running Courier
+    fi
+
 	;;
 restart)
 	$0 stop
