--- courier-0.35.1/rootcerts/configure.in.orig	Sun May 27 05:02:12 2001
+++ courier-0.35.1/rootcerts/configure.in	Thu Jan  3 12:40:51 2002
@@ -10,7 +10,7 @@
 AM_INIT_AUTOMAKE(rootcerts, 0.50)
 
 AC_PATH_PROG(OPENSSL, openssl, "", $PATH:/sbin:/usr/sbin:/usr/local/sbin:/usr/local/bin)
-AC_PATH_PROG(REHASH, c_rehash, "", $PATH:/sbin:/usr/sbin:/usr/local/sbin:/usr/local/bin)
+AC_PATH_PROG(REHASH, c_rehash, "", $PATH:/sbin:/usr/sbin:/usr/local/sbin:/usr/local/bin:/usr/lib/openssl)
 AC_PATH_PROGS(PERL, perl5 perl, "", $PATH:/usr/local/bin)
 
 if test "$PERL" = ""
