diff -Nur old/authlib/configure.in new/authlib/configure.in
--- old/authlib/configure.in	2004-03-03 04:45:09.000000000 +0000
+++ new/authlib/configure.in	2004-03-16 21:35:08.000000000 +0000
@@ -385,36 +385,6 @@
 	LIBS="$NETLIBS $LIBS"
 	LDAPLIBS=""
 
-        AC_MSG_CHECKING(whether -lresolv is needed for res_query)
-	AC_LINK_IFELSE(AC_LANG_PROGRAM([
-#if HAVE_NETINET_IN_H
-#include <netinet/in.h>
-#endif
-#include <resolv.h>
-
-void (*func)()= (void (*)())res_query;
-],
-			[  (*func)(); ]),
-        	AC_MSG_RESULT(no),
-		[
-        	LIBS="-lresolv $LIBS"
-
-		AC_LINK_IFELSE(AC_LANG_PROGRAM([
-#if HAVE_NETINET_IN_H
-#include <netinet/in.h>
-#endif
-#include <resolv.h>
-
-void (*func)()= (void (*)())res_query;
-],
-			[  (*func)(); ]), [
-        		LDAPLIBS="-lresolv $LDAPLIBS"
-			HAVE_LDAP=1
-        		AC_MSG_RESULT(yes)
-			],
-        		AC_MSG_ERROR(Cannot find function res_query))
-			])
-
 	AC_CHECK_LIB(lber, ber_scanf,
 		[ LDAPLIBS="-llber $LDAPLIBS" LIBS="-llber $LIBS" ])
 	AC_CHECK_LIB(ldap, ldap_open,
diff -Nur old/courier/configure.in new/courier/configure.in
--- old/courier/configure.in	2004-03-07 04:39:48.000000000 +0000
+++ new/courier/configure.in	2004-03-16 21:35:21.000000000 +0000
@@ -251,38 +251,6 @@
 AC_CHECK_LIB(lber, ber_scanf,
 	[ LDAPLIBS="-llber" LIBS="-llber $LIBS" ])
 
-
-        AC_MSG_CHECKING(whether -lresolv is needed for res_query)
-	AC_LINK_IFELSE(AC_LANG_PROGRAM([
-
-#if HAVE_NETINET_IN_H
-#include <netinet/in.h>
-#endif
-#include <resolv.h>
-
-void (*func)()= (void (*)())res_query;
-],
-			[  (*func)(); ]),
-        	AC_MSG_RESULT(no),
-		[
-        	LIBS="-lresolv $LIBS"
-
-		AC_LINK_IFELSE(AC_LANG_PROGRAM([
-#if HAVE_NETINET_IN_H
-#include <netinet/in.h>
-#endif
-#include <resolv.h>
-
-void (*func)()= (void (*)())res_query;
-],
-			[  (*func)(); ]), [
-        		LDAPLIBS="-lresolv $LDAPLIBS"
-			HAVE_LDAP=1
-        		AC_MSG_RESULT(yes)
-			],
-        		AC_MSG_ERROR(Cannot find function res_query))
-			])
-
 AC_CHECK_LIB(ldap, ldap_open,
 	[ LDAPLIBS="-lldap $LDAPLIBS" ; LIBS="-lldap $LIBS" ])
 AC_CHECK_FUNC(ldap_search_st, HAVE_LDAP=1, HAVE_LDAP=0)
