diff -Nur old/courier/imapd.rc.in new/courier/imapd.rc.in
--- old/courier/imapd.rc.in	2003-11-29 02:16:15.000000000 +0000
+++ new/courier/imapd.rc.in	2004-05-24 21:23:33.000000000 +0000
@@ -53,6 +53,7 @@
 	;;
 stop)
 	@sbindir@/couriertcpd -pid=$PIDFILE -stop
+	rm -f $PIDFILE
 	;;
 restart)
 	@sbindir@/couriertcpd -pid=$PIDFILE -restart
diff -Nur old/courier/imapd-ssl.rc.in new/courier/imapd-ssl.rc.in
--- old/courier/imapd-ssl.rc.in	2003-11-29 02:16:15.000000000 +0000
+++ new/courier/imapd-ssl.rc.in	2004-05-24 21:25:19.000000000 +0000
@@ -54,6 +54,7 @@
 	;;
 stop)
 	@sbindir@/couriertcpd -pid=$SSLPIDFILE -stop
+	rm -f $SSLPIDFILE
 	;;
 esac
 exit 0
diff -Nur old/courier/module.esmtp/esmtpd.in new/courier/module.esmtp/esmtpd.in
--- old/courier/module.esmtp/esmtpd.in	2004-05-11 00:48:14.000000000 +0000
+++ new/courier/module.esmtp/esmtpd.in	2004-05-24 21:23:33.000000000 +0000
@@ -38,6 +38,7 @@
 	;;
 stop)
 	${sbindir}/couriertcpd -pid=$PIDFILE -stop
+	rm -f $PIDFILE
 	exit 0
 	;;
 restart)
diff -Nur old/courier/module.esmtp/esmtpd-ssl.in new/courier/module.esmtp/esmtpd-ssl.in
--- old/courier/module.esmtp/esmtpd-ssl.in	2004-05-11 00:48:14.000000000 +0000
+++ new/courier/module.esmtp/esmtpd-ssl.in	2004-05-24 21:24:41.000000000 +0000
@@ -28,6 +28,7 @@
 	;;
 stop)
 	${sbindir}/couriertcpd -pid=$SSLPIDFILE -stop
+	rm -f $SSLPIDFILE
 	exit 0
 	;;
 restart)
diff -Nur old/courier/pop3d.in new/courier/pop3d.in
--- old/courier/pop3d.in	2003-11-28 01:04:59.000000000 +0000
+++ new/courier/pop3d.in	2004-05-24 21:23:33.000000000 +0000
@@ -42,6 +42,7 @@
 	;;
 stop)
 	${sbindir}/couriertcpd -pid=$PIDFILE -stop
+	rm -f $PIDFILE
 	exit 0
 	;;
 restart)
diff -Nur old/courier/pop3d-ssl.in new/courier/pop3d-ssl.in
--- old/courier/pop3d-ssl.in	2003-11-29 01:37:34.000000000 +0000
+++ new/courier/pop3d-ssl.in	2004-05-24 21:25:31.000000000 +0000
@@ -58,6 +58,7 @@
 	;;
 stop)
 	@sbindir@/couriertcpd -pid=$SSLPIDFILE -stop
+	rm -f $SSLPIDFILE
 	;;
 restart)
 	@sbindir@/couriertcpd -pid=$SSLPIDFILE -restart
diff -Nur old/courier.sysvinit.in new/courier.sysvinit.in
--- old/courier.sysvinit.in	2004-05-02 14:24:06.000000000 +0000
+++ new/courier.sysvinit.in	2004-05-24 21:23:33.000000000 +0000
@@ -8,6 +8,9 @@
 #
 #
 
+# Source function library
+. /etc/rc.d/init.d/functions
+
 prefix="@prefix@"
 exec_prefix="@exec_prefix@"
 sysconfdir="@sysconfdir@"
@@ -24,6 +27,11 @@
 
 case "$1" in
 start)
+	if [ -e /var/lock/subsys/courier ]; then
+	    msg_already_running Courier
+	    exit 0
+	fi
+
         cd /
         # Start daemons.
         touch /var/lock/subsys/courier
@@ -196,6 +204,9 @@
 	echo ""
         ;;
 stop)
+    if [ -e /var/lock/subsys/courier ]; then
+
+        rm -f /var/lock/subsys/courier
         echo -n "Stopping Courier mail server:"
 
 	if test -x ${sbindir}/imapd
@@ -254,6 +265,10 @@
 	fi
 	${sbindir}/courierfilter stop
 	echo " courierfilter"
+
+    else
+	    msg_not_running Courier
+    fi
 	;;
 restart)
 	$0 stop
