--- courier-0.78.2.orig/courier/courierd.dist.in	2009-11-08 19:14:47.000000000 +0100
+++ courier-0.78.2/courier/courierd.dist.in	2010-05-08 23:27:21.826603250 +0200
@@ -88,9 +88,9 @@
 #  One of the following definitions of DEFAULTDELIVERY should be
 #  uncommented.
 #
-#  Default deliveries to $HOME/Maildir
+#  Default deliveries to $HOME/Mail/Maildir
 #
-#  DEFAULTDELIVERY=./Maildir
+#  DEFAULTDELIVERY=./Mail/Maildir
 #
 #  Alternatively, use procmail to deliver mail to local mailboxes.
 #
@@ -104,12 +104,12 @@
 #  use something like this:
 #
 #  DEFAULTDELIVERY="|| dotforward
-#  ./Maildir"
+#  ./Mail/Maildir"
 #
 #  Yes, it's two lines long, with an embedded newline.  Of course, you can use
 #  any default local mail delivery instruction in place of ./Maildir.
 
-DEFAULTDELIVERY=./Maildir
+DEFAULTDELIVERY=./Mail/Maildir
 
 ##NAME: MAILDROPDEFAULT:0
 #
@@ -117,7 +117,7 @@
 #  the location of the default mailbox.  You should not change this setting
 #  unless you REALLY know what you're doing.
 
-MAILDROPDEFAULT=./Maildir
+MAILDROPDEFAULT=./Mail/Maildir
 
 ##NAME: ESMTP_CORK:0
 #
--- courier-0.78.2.orig/libs/imap/imapd.dist.in	2008-06-21 18:01:23.000000000 +0200
+++ courier-0.78.2/libs/imap/imapd.dist.in	2010-05-08 23:27:21.829937594 +0200
@@ -417,4 +417,4 @@
 #
 # MAILDIRPATH - directory name of the maildir directory.
 #
-MAILDIRPATH=Maildir
+MAILDIRPATH=Mail/Maildir
--- courier-0.78.2.orig/libs/imap/imapd-ssl.dist.in	2009-08-13 00:25:49.000000000 +0200
+++ courier-0.78.2/libs/imap/imapd-ssl.dist.in	2010-05-08 23:27:21.829937594 +0200
@@ -329,4 +329,4 @@
 #
 # MAILDIRPATH - directory name of the maildir directory.
 #
-MAILDIRPATH=Maildir
+MAILDIRPATH=Mail/Maildir
--- courier-0.78.2.orig/libs/imap/pop3d.dist.in	2005-07-05 14:42:51.000000000 +0200
+++ courier-0.78.2/libs/imap/pop3d.dist.in	2010-05-08 23:27:21.829937594 +0200
@@ -143,4 +143,4 @@
 #
 # MAILDIRPATH - directory name of the maildir directory.
 #
-MAILDIRPATH=Maildir
+MAILDIRPATH=Mail/Maildir
--- courier-0.78.2.orig/libs/imap/pop3d-ssl.dist.in	2009-08-13 00:25:49.000000000 +0200
+++ courier-0.78.2/libs/imap/pop3d-ssl.dist.in	2010-05-08 23:27:21.835978217 +0200
@@ -315,4 +315,4 @@
 #
 # MAILDIRPATH - directory name of the maildir directory.
 #
-MAILDIRPATH=Maildir
+MAILDIRPATH=Mail/Maildir
--- courier-0.78.2.orig/libs/pcp/pcpd.c	2006-01-24 00:22:10.000000000 +0100
+++ courier-0.78.2/libs/pcp/pcpd.c	2010-05-08 23:27:21.835978217 +0200
@@ -2187,10 +2187,10 @@
 		exit(1);
 	}
 
-	if (chdir(a->maildir && *a->maildir ? a->maildir:"Maildir") < 0)
+	if (chdir(a->maildir && *a->maildir ? a->maildir:"Mail/Maildir") < 0)
 	{
 		free(token);
-		fprintf(stderr, "NOTICE: chdir(Maildir) failed: %s\n",
+		fprintf(stderr, "NOTICE: chdir(Mail/Maildir) failed: %s\n",
 			strerror(errno));
 		maildir_cache_cancel();
 		exit(1);
--- courier-0.78.2.orig/libs/sqwebmail/auth.c	2009-06-27 18:32:50.000000000 +0200
+++ courier-0.78.2/libs/sqwebmail/auth.c	2010-05-08 23:27:21.835978217 +0200
@@ -79,7 +79,7 @@
 	if (!maildir || !*maildir)
 		maildir=getenv("MAILDIRPATH");
 	if (!maildir || !*maildir)
-		maildir="Maildir";
+		maildir="Mail/Maildir";
 	if (chdir(maildir))	return (-1);
 	maildir_loginexec();
 	return (0);
--- courier-0.78.2.orig/libs/sqwebmail/sqwebmaild.dist.in	2007-02-10 04:16:35.000000000 +0100
+++ courier-0.78.2/libs/sqwebmail/sqwebmaild.dist.in	2010-05-08 23:27:21.835978217 +0200
@@ -32,7 +32,7 @@
 #
 # MAILDIRPATH - directory name of the maildir directory.
 
-MAILDIRPATH=Maildir
+MAILDIRPATH=Mail/Maildir
 
 ##NAME: SQWEBMAIL_PREFORK:0
 #
--- courier-0.78.2.orig/libs/sqwebmail/sqwebmail.h	2007-02-24 20:02:59.000000000 +0100
+++ courier-0.78.2/libs/sqebmail/sqwebmail.h	2010-05-08 23:27:21.835978217 +0200
@@ -25,7 +25,7 @@
 
 /* Location of the user's Maildir */
 
-#define USER_DIR	"Maildir"
+#define USER_DIR	"Mail/Maildir"
 
 /* For PAM-based authentication */
 
