diff -Nur old/courier/Makefile.am new/courier/Makefile.am
--- old/courier/Makefile.am	2004-04-01 23:48:27.000000000 +0000
+++ new/courier/Makefile.am	2004-04-10 21:00:03.000000000 +0000
@@ -62,8 +61,8 @@
 sbinscripts=sharedindexsplit sharedindexinstall
 
 sbin_SCRIPTS=@SBINSCRIPTS@
-
+EXTRA_SCRIPTS=
 sbin_PROGRAMS=courier showconfig showmodules sendmail \
 	@COURIERLDAPALIASD@ @EXTRASBIN@
 
 sysconf_DATA=pop3d.authpam imapd.authpam webmail.authpam imapd.cnf pop3d.cnf \
